src/utils/process_info.rs

```diff
@@ -10,45 +8,31 @@ const TARGET_PROCESS: &str = "LeagueClientUx.";
 #[cfg(target_os = "macos")]
 const TARGET_PROCESS: &str = "LeagueClientUx";
 
-pub fn find_process(system: &System) -> Result<String, &'static str> {
-    system
+pub fn get_auth_info() -> Result<(String, u16), &'static str> {
+    let mut sys = System::new_all();
+    sys.refresh_processes();
+
+    let args = sys
         .processes()
         .values()
-        .find(|process| process.name() == TARGET_PROCESS)
-        .map(|process| process.cmd().join(" "))
-        .ok_or("Could not find a running LCU process!")
-}
-
-lazy_static! {
-    static ref PORT_RE: Regex = Regex::new(r"--app-port=([0-9]*)").unwrap();
-    static ref TOKEN_RE: Regex = Regex::new(r"--remoting-auth-token=([\w-]*)").unwrap();
-}
+        .find(|p| p.name() == TARGET_PROCESS)
+        .map(|p| p.cmd())
+        .ok_or("no LoL process")?;
 
-pub fn extract_info(cmd_args: String) -> Result<(String, u32), &'static str> {
-    let port: u32 = PORT_RE
-        .captures(&cmd_args)
-        .map_or(Err("No port found!"), |value| {
-            value.get(1).map_or(Err("No port found!"), |port| {
-                port.as_str()
-                    .parse()
-                    .map_or(Err("Failed to parse port"), |port| Ok(port))
-            })
-        })?;
-
-    let token =
-        TOKEN_RE
-            .captures(&cmd_args)
-            .map_or(Err("No authentication token found!"), |value| {
-                value
-                    .get(1)
-                    .map_or(Err("No authentication token found!"), |token| {
-                        Ok(token.as_str())
-                    })
-            })?;
-
-    Ok((token.to_string(), port))
-}
+    let port = args
+        .into_iter()
+        .find(|arg| arg.starts_with("--app-port="))
+        .map(|arg| arg[11..].parse::<u16>().ok())
```

Comment:
Is there a reason this is parsed to an u16?

Reply:
port numbers fit exactly into an u16, if the number is outside of that range it is invalid

Reply:
Right, but it doesn't actually need to be stored as a u16, and will remove the parse step to store it as a string